### DISCUSSION TITLE
👋🏽 New here? Introduce yourself!

### DISCUSSION THREAD

**Discussion by leouieda** [fatiando/community#55] [Announcements]
If you are new to the Fatiando discussion forum, please **feel free to introduce yourself in this thread**. We are all humans and it's nice to get to know the people in our community ❤️ 

You could **just say "Hi"** or tell us a **bit about yourself**. Which ever feels right for you.

**Comment by leouieda** [comment]
Well, I guess I should start... 🙂 

👋🏾 Hi everyone! I'm Leo and I am one of the original creators of Fatiando (way back in 2010 during my MSc in Geophysics). I'm from Brazil 🇧🇷 but I'm currently in the UK 🇬🇧 working as Lecturer (UK name for Assistant Professor) in Geophysics at the University of Liverpool. I use Fatiando for basically all of my research and much of my teaching. I also work on other geo-related Python software, like [PyGMT](https://www.pygmt.org/).

To find out more, head on over to my personal website: https://www.leouieda.com

**Comment by santisoler** [comment]
Hola! Hi everyone!

I'm Santiago, Physicist and PhD in Geophysics from Argentina 🇦🇷. I started using Fatiando during my Licentiate Thesis back in 2014 and getting more and more involved in the project in the following years. Fatiando was a major backbone of my PhD Thesis: my research made heavy usage of it while the new methods developed during it live now inside Harmonica. I loved this feedback of science and open-source software, but mostly I loved how science can move a lot faster through these kind of collaborations.

In the near future I'll start a Postdoctoral Research Fellow position at University of British Columbia, in which I will work not only with Fatiando but also with [SimPEG](https://simpeg.xyz/). I'm looking forward to see what will come out of that!

Read more about me in [www.santisoler.com](https://www.santisoler.com)

**Comment by MGomezN** [comment]
>A long time ago, in an institution far,
 far away... 🇲🇽 

_**Episode IV
A NEW HOPE**_

It is a period of Ph.D. The ultimate goal is to develop strong programming skills while doing hydrogeophysics on the Bahamas and The Peninsula of Yucatan, Mexico. 
During the battle, Mariana joined Fatiando a Terra to learn open-source plans to transform science into a more justice place to generate knowledge. 
Pursued by the deadlines of Academia, Mariana races all over the place as custodian of the best practices of teaching that can save her people and restore freedom to the Center for Scientific Research and Higher Education at Ensenada…

>Jaja ok, a mythomaniac here, jaja No, Someone who tries to enjoy and laugh every time she cans :)

**Reply by leouieda** [reply]
Did I already mention that I love your writing? Happy to contribute to your fight for freedom (of code)!

**Reply by santisoler** [reply]
Awesome to have you on board, it is.

**Comment by LL-Geo** [comment]
大家好! Hello everyone!

I'm Lu, a geophysicist from China 🇨🇳. I am doing my PhD in Australia 🇦🇺, modelling the lithosphere structure of Antarctica🐧, and understanding the interaction between solid-earth and cryosphere. It's quite challenging but really fun!!

I love to talk about ideas within the Fatiando community. It's really great everyone could work together to benefit the whole community. Also, I think Fatiando has a great ability and potential to solve exciting challenges!!! (Looks like it's a good time to build my personal website, haha 😆)

**Reply by randy7771026** [reply]
OK so maybe you know me but likely not and I really have no idea of Fatiando but love python and used to do Fortran and PL/1 for Dorman, Levin and Vail playing with basement, VSP, Love, Raleigh and Shear.

**Reply by leouieda** [reply]
@randy7771026 playing around with simulations of Love and Raleigh waves was one of the first drivers for me to work on Fatiando. The old `fatiando` package (now discontinued) had a finite-difference seismic wave simulation that could generate videos and embed them directly into Jupyter notebooks. Here is an example using that code in one of my classes from back in 2015: https://nbviewer.ipython.org/github/leouieda/geofisica2/blob/master/notebooks/2-reflexao-refracao.ipynb

The code underlying all of that is here: https://github.com/fatiando/fatiando/tree/wavefd/fatiando/seismic/wavefd I always meant to strip that out, updated it, and make into an educational package but never got around to, particularly since I don't teach seismics these days.

**Comment by randy7771026** [comment]
Very cool.  Me espanol no es bueno.  I picked up enough in the notebook to
see the cool though,  Jupyter has also been a problem for me with version
control.  Jim Dorman is who I was coding with on the long seismic waves and
some in group were interested in wave conversions.

I will have to see if I have time to set up and run Jupyter and delve into
what fatiando does in well logs as well.

The local SEG was playing with old sismic datasets a few years back,
precovid and I would run into one of their main catalogers at visualization
meetups.

**Comment by hwil055** [comment]
Kia Ora,
I'm a part-time lecturer at Auckand University of Technology but also part-time consultant to the Minerals Exploration industry. I have  a background/career in geophysics but am responsible for educating new generation of Earth Scientists. Am keenly aware of value of computational geoscience but need to train (often, not always) underprivileged audience. Love python, love open-source, just not as competent as I'd like to be....
Probably confusing - apologies.....

**Reply by leouieda** [reply]
Kia Ora, @hwil055! Welcome to Fatiando 👋🏾 

Many of us here are also involved in compgeo education so this is good place to share resources and ask questions. The initial motivation I had for starting Fatiando was trying to educate myself in different aspects of geoscience and I used it extensively in my teaching. I know @santisoler @aguspesce and @andieie  were recently involved in [Code to Communicate](https://github.com/CodeToCommunicate/) (they even had a Fatiando tutorial) so there is plenty of common ground 🙂

**Reply by hwil055** [reply]
Obrigado 😃

**Comment by menarinis** [comment]
Hi, I'm Stefano Menarini from Italy and I work in agriculture. I discovered Verde after a research on how to decimate the points acquired with gps. I have this need for decimation. I acquire land metrics data on fields of a few hectares each time. I need to decimate my acquired positions along generally straight and parallel lines (the path of a tractor). I experimented with Verde's datasets and I understood how it works. Now I would like to use it with my gps data which is in text format (point name, latitude, longitude). I ask you can I use Green to decimate my gps points? If yes, how can I import my file with the data in Green and follow the decimation procedure? Thanks a lot in advance for the answer.

**Reply by MGomezN** [reply]
Hello, @menarinis ! 

Welcome to the Fatiando Community!
Thank you for your question. 

Maybe this video tutorial from 2020 is helpful for you. It shows  the process of loading files (time stamp 17:09), and the "decimation"in Verde we called it "Block reductions" (36:30)

https://www.youtube.com/live/-xZdNdvzm3E?feature=share

You can load your data file using the Python Pandas library, for example: 
```
df=pd.read_csv('file.txt')
```

Let us know if it worked. We are happy to help :)

**Reply by MGomezN** [reply]
@menarinis
Addendum. 
Verde decimates data by applying a reduction operation (mean, median, mode, max, etc)to the data in blocks.
Your txt file has (point name, latitude, longitude) but doesn't have another column with the physical parameter (e.g., bathymetry),  right? 

So, I think you will need to create an artificial extra column,  filled with a random value (ones, for example),  that allows you perform the block reduction operation

```
df['extra_column'] = 1
```

**Reply by menarinis** [reply]
Thank you all. I managed to import some files without problems by transforming them into csv with comma separator and adding an extra column as you suggested. I followed the video to practice and arrived, using my data in Jupyter, at minute 32.13. Now I can't go on here. My data is already in UTM coordinates 33N and in the proj documentation I can't find information about the syntax part of Pyproj-proj ...lat_ts=data...
from which I get error. In the video it seems that the author wants to transform geographical coordinates or am I wrong? How can I move forward from here?
I also proceeded in this way with the following commands following your tutorial (https://www.fatiando.org/verde/latest/tutorials/decimation.html):
reducer = vd.BlockReduce(reduction=np.median, spacing=5 / 60) plt.figure(figsize=(7, 7))
ax = plt.axes(projection=ccrs.utm(33, southern_hemisphere=False))
ax.set_title("Locations of decimated data")
# Plot the bathymetry data locations as black dots
plt.plot(*coordinates, ".k", markersize=1, transform=crs)
but I get the error: ...AttributeError: module 'cartopy.crs' has no attribute 'utm'...
I would like to complete, without errors, the tutorial up to saving the reduced data file. Thanks in advance

**Comment by MGomezN** [comment]
Hi @menarinis 
Sorry for the reply until now, 
If you already have a dataframe `data` with columns `easting` and `northing` , then I don't see why you want to project again with cartopy or pyproj. You can skip that part and you can directly define your coordinates like this:

`coordinates= (data['easting'].to_numpy(), data['northing'].to_numpy())`

So, complete with my previous suggestions, taken from the tutorial https://www.youtube.com/watch?v=-xZdNdvzm3E it will look like this:

```
data = pd.read_csv('file.txt')
data['extra_column'] = 1
coordinates= (data['easting'].to_numpy(), data['northing'].to_numpy())`

reducer = vd.BlockReduce(np.median, spacing=20)

block_coords, block_bathymetry = reducer.filter(coordinates, data.extra_column)
data.shape

block_bathymetry.shape

plt.figure()
plt.scatter(block_coords[0], block_coords[1], c=block_bathymetry, s=1)
plt.axis("scaled")
plt.colorbar()
```
Let me know if this was helpful, we can also coordinate a meeting in case it is easier to share your screen.

**Comment by Phssilva** [comment]
Hello, I'm from Brazil and I'm a geology undergraduate student at the Federal University of Santa Catarina. I found this project while researching geophysics and Python topics and would love to contribute improvements.

I started working as a programmer in 2009 and have worked in some financial institutions with databases and cloud computing. Currently, I work as a data engineer on a research and development project for geo-environmental monitoring, using artificial intelligence for energy transmission line monitoring.

I'm now starting to contribute to open-source projects.
I found this project very interesting and would like to contribute more.

**Reply by leouieda** [reply]
Bem-vindo @Phssilva! Your help is much appreciated ❤️

**Comment by strainflowandmore** [comment]
Hey! I am Alex from New Mexico Tech in the USA. I am a hydrology prof here who works in a range of things, including doing repeat gravity surveys to look at changing water storage. But, I am getting into working on regional potential field problems and am interested in using Fatiando a Terra for our forward and inverse modeling of extensional basin structure.

I can run the surveys, but haven't done complicated inversions of this type of data yet, so I am looking forward to learning from ya'll on using the programs.

**Reply by santisoler** [reply]
Hi @strainflowandmore! Welcome to the community! 👋🏼
